web/src/components/PopUpForm.js

```diff
@@ -0,0 +1,88 @@
+import React, {useState} from 'react';
+import ValueInputer from "./ValueInputer";
+import '../styles/CreateGroup.css';
+import { useUser } from "../providers/UserProvider";
+
+/**
+ * Function that creates a popup form for a user to input
+ * information regarding preferences of a group to create
+ * or search for.
+ *
+ * @param props must contain:
+ *  onChange: function to call when popup form is submitted
+ *  maxGroupSize: constant for maximum group size
+ *  action: value to display on the submit button
+ */
+export default function PopUpForm(props) {
+    const [maxGroupSize, setMaxGroupSize] = useState(0);
+    const [groupName, setGroupName] = useState("");
+    const [classPrefix, setClassPrefix] = useState("");
+    const [classNum, setClassNum] = useState(0);
+    const [classSection, setClassSection] = useState("");
+    const [topics, setTopics] = useState(0);
+    const [meetInPerson, setMeetInPerson] = useState(false);
+    const user = useUser.user;
+
+    function closeForm() {
+        props.onChange("", "", "", "", "", "");
+    }
+
+    function recordData() {
+        // ***** VALIDATE INPUT *****
+        if (maxGroupSize <= 1) {
+            alert("Enter a valid group size (must be greater than 1).");
+            return;
+        }
+        if (groupName === "" || groupName === null) {
+            alert("Enter a valid group name.");
+            return;
+        }
+
+        if (classPrefix === "" || groupName === null) {
+            alert("Enter a valid class prefix (Example: CSE, ENGL, MATH, etc).");
+            return;
+        }
+
+        if (classNum <= 0 || classNum > 1000) {
```

Comment:
might be good to report this restriction to the user (e.g. "enter a class number between 0 and 1000").

Reply:
Also - the error reporting in this module might be slightly improved if you built up a list of all the different errors, if thee were more than 1, so the user doesn't have to fix one at a time